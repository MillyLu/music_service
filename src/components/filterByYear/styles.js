
import styled from "styled-components";



export const Year = styled.div`
position: absolute;
left: 597px;
top: 288px;
width: 403px;
height: 92px;
background: ${props => props.theme === "light" ? "#FFFFFF" : "#313131"}; 
border: ${props => props.theme === "light" ? "2px solid #AD61FF" : 'none'}; 
border-radius: 12px;
display: flex;
flex-direction: row;
align-items: flex-start;
justify-content: center;
padding: 34px;
gap: 24px;
`;

export const YearInput = styled.input`
width: 0;
height: 0;
position: absolute;
opacity: 0;
z-index: -1;
`;

export const Label = styled.label``;



export const Item = styled.div`
  display: flex;
  align-items: center;
  position: relative;
  box-sizing: border-box;
`;

export const RadioButtonLabel = styled.label`
  position: absolute;
  top: 25%;
  left: 4px;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  background: black;
  border: 1px solid #ccc;
`;
export const RadioButton = styled.input`
  opacity: 0;
  z-index: 1;
  cursor: pointer;
  width: 20px;
  height: 20px;
  margin-right: 10px;
  &:hover ~ ${RadioButtonLabel} {
    background: white;
    &::after {
      content: "\f111";
      font-family: "FontAwesome";
      display: block;
      color: white;
      width: 10px;
      height: 10px;
      margin: 3px;
    }
  }
  &:checked + ${Item} {
    background: black;
    border: 2px solid black;
  }
  &:checked + ${RadioButtonLabel} {
    background: black;
    border: 1px solid black;
    &::after {
      content: "\f111";
      font-family: "FontAwesome";
      display: block;
      color: ${props => props.theme === "light" ? "#000000" : "#FFFFFF"};
      width: 10px;
      height: 10px;
      margin: 3px;
    }
  }
`;
