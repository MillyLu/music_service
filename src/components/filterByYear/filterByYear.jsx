import { useState, useEffect, useContext } from 'react';
import * as Styled from './styles';
import { ThemeContext } from '../../ThemeProvider';

export function FilterButtonYear(props) {  


    
    const [year,setYear] = useState();

    const changeChoose = (e) => {

        setYear(e.target.value)
    }

    useEffect(() => {
        console.log(year);
        props.getTracksByData(year)

    }, [changeChoose]);

    const {theme} = useContext(ThemeContext);
    return(
        <Styled.Year theme={theme}>
            <Styled.Item>
                <Styled.RadioButton
                theme={theme}
                    type="radio"
                    name="year"
                    value="new"
                    checked={year==="new"}
                    onChange={changeChoose}
                  
                />
                <Styled.RadioButtonLabel />
                <div style={{color:theme === "light" ? "#000000" : "#FFFFFF"}}>Более новые</div>
            </Styled.Item>
            <Styled.Item>
                <Styled.RadioButton
                theme={theme}
                type="radio"
                name="year"
                value="old"
                checked={year==="old"}
                onChange={changeChoose}
                 />
                <Styled.RadioButtonLabel />
                <div style={{color:theme === "light" ? "#000000" : "#FFFFFF"}}>Более старые</div>
            </Styled.Item>
            
        </Styled.Year>
    )
}
