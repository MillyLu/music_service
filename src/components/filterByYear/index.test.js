/* eslint-disable no-undef */


import * as ShallowRenderer from 'react-shallow-renderer';
import { FilterButtonYear } from './filterByYear';




describe('<FilterButtonYear/>', () => {
        it('should render successfully and match snapshot', () => {

             const shallowRenderer = ShallowRenderer.createRenderer();
             const renderer = shallowRenderer;
             renderer.render(<FilterButtonYear />);
             const result = renderer.getRenderOutput();
             expect(result).toMatchSnapshot();  

          })
      });