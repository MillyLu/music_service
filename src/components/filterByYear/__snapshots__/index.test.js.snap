// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButtonYear/> should render successfully and match snapshot 1`] = `
<styled.div
  theme={0}
>
  <styled.div>
    <styled.input
      checked={false}
      name="year"
      onChange={[Function]}
      theme={0}
      type="radio"
      value="new"
    />
    <styled.label />
    <div
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Более новые
    </div>
  </styled.div>
  <styled.div>
    <styled.input
      checked={false}
      name="year"
      onChange={[Function]}
      theme={0}
      type="radio"
      value="old"
    />
    <styled.label />
    <div
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Более старые
    </div>
  </styled.div>
</styled.div>
`;
